### ISSUE TITLE
[WS] Currency translatable fields empty in webservice 1.7.6.1

### ISSUE THREAD

**Issue by Perfecticus** [PrestaShop/PrestaShop#16760]
**Describe the bug**

When listing currencies using the web service, the translatable fields are empty.

Also when updating currencies using web service, the translatable fields are cleared/emptied even if no **name** or **symbol** element is present (removed from xml).

This has implications when doing patch-work using external services.
For example loading existing currency data, re-populating currency rate and then pushing it back to prestashop. All translatable fields are emptied.

**To Reproduce**

Browse your currencies endpoint:
yourdomain.com/api/currencies?display=full

**Screenshots**

![Before PUT](https://puu.sh/ENV03/81b82c585b.png)

![Web service response](https://puu.sh/ENUQf/42fa307078.png)

![Example GET/PUT](https://puu.sh/ENUWq/88d58a3afc.png)

![Cleared tables after PUT](https://puu.sh/ENUTr/c0a92cc410.png)

**Additional information**
PrestaShop version: 1.7.6.1 release
PHP version: 7.0.33

**Comment by khouloudbelguith** [comment]
Hi @Perfecticus,

Thanks for your report.
>When listing currencies using the web service, the translatable fields are empty.

I manage to reproduce the issue with PS1.7.6.2 & PS1.7.6.0
![image](https://user-images.githubusercontent.com/16067358/70603937-ede06680-1bf7-11ea-8487-aa25d70d1b92.png)
With PS1.7.5.2, it was ok
![image](https://user-images.githubusercontent.com/16067358/70603982-06e91780-1bf8-11ea-9e98-5d8fb2bd12f1.png)
It is a regression.

>Also when updating currencies using web service, the translatable fields are cleared/emptied even if no name or symbol element is present (removed from xml).

I tried to update currency using this script
```php
<html><head><title>pUT DATA</title></head><body>
<?php

define('DEBUG', true);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'shop.com');
define('PS_WS_AUTH_KEY', 'key');
require_once('PSWebServiceLibrary.php');
$ws = new PrestaShopWebservice(
    'shop.com',
    'Key',
    false);

$id = 1; //ID currency Here
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "BEFORE\n";
echo $xml->asXML();
unset($xml->currency->symbol);
$xml->currency->name->language[0][0] = "edit euro";
$opt['putXml'] = $xml->asXML();
$opt['id'] = $id;
$opt['resource'] = 'currencies';
$xml = $ws->edit($opt);
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "AFTER\n";
echo $xml->asXML();
```
With ps1.7.6.2 & PS1.7.6.0,
The table ps_currency_lang well updated => but all other fiellds will be empty
**Before update**
ps_currency_lang
![image](https://user-images.githubusercontent.com/16067358/70604307-bde59300-1bf8-11ea-8218-9e2d5ae9905b.png)
Currencies page
![image](https://user-images.githubusercontent.com/16067358/70604478-09983c80-1bf9-11ea-9d1a-661465b0ffa7.png)

**After Update**, I have this
ps_currency_lang
![image](https://user-images.githubusercontent.com/16067358/70604624-39474480-1bf9-11ea-8fd3-1d78bac4dacf.png)
Currencies page
![image](https://user-images.githubusercontent.com/16067358/70604663-4ebc6e80-1bf9-11ea-9bb1-cc63894c42db.png)

With PS1.7.5.2, the table ps_currency_lang doesn't exist & it is not possible to update the name via web service, I used this script
```php
<html><head><title>pUT DATA</title></head><body>
<?php
define('DEBUG', true);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'shop.com');
define('PS_WS_AUTH_KEY', 'key');
require_once('PSWebServiceLibrary.php');
$ws = new PrestaShopWebservice(
    'shop.com',
    'Key',
    false);
$id = 2; //ID specific rule Here
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "BEFORE\n";
echo $xml->asXML();
unset($xml->currency->symbol);
$xml->currency->name= "EDIT Russian Ruble";
$opt['putXml'] = $xml->asXML();
$opt['id'] = $id;
$opt['resource'] = 'currencies';
$xml = $ws->edit($opt);
$xml = $ws->get(array('resource' => 'currencies', 'id' => $id));
echo "AFTER\n";
echo $xml->asXML();
```
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by Progi1984** [comment]
Fixed in #17069
